### PR TITLE
Clean stacking when unbonding all values (#154)

### LINKED ISSUE #154: How to Clean Staking System?
The `withdraw_unbonded` was removed.

How to trigger the `kill_stash` to remove the 'empty' account and keep the staking system clean?

```rust
StakingLedger {
	active_ring/kton: zero,
	ring/kton_lock: all expired,
	...
}
```

### PR DESCRIPTION
**Pull request by yanganto**
- call `kill_stash` when all values are unbonded
- fix #154

### CODE REVIEW

File: srml/staking/src/lib.rs [hunk 1]
```diff
@@ -877,6 +882,22 @@ decl_module! {
 					}
 				},
 			}
+
+			let updated_ledger = Self::clear_mature_deposits(
+				Self::ledger(&controller).ok_or(err::CONTROLLER_INVALID)?);
+
+			let stash = ledger.stash.clone();
```
Comment (AurevoirXavier) [inline srml/staking/src/lib.rs]: unnecessary `clone`

File: srml/staking/src/lib.rs [hunk 2]
```diff
@@ -877,6 +882,22 @@ decl_module! {
 					}
 				},
 			}
+
+			let updated_ledger = Self::clear_mature_deposits(
```
Comment (AurevoirXavier) [inline srml/staking/src/lib.rs]: `active_ring = active_normal_ring + active_deposit_ring` useless checking

File: srml/staking/src/lib.rs [hunk 3]
```diff
@@ -877,6 +882,22 @@ decl_module! {
...
+			let stash = ledger.stash.clone();
+			let StakingLedger {
+				active_ring,
+				active_deposit_ring,
+				active_kton,
+				..
+			} = updated_ledger;
+
+			// all bonded rings and ktons is withdrawing, then remove Ledger to save storage
+			if active_ring.is_zero() && active_deposit_ring.is_zero() && active_kton.is_zero() {
```
Comment (AurevoirXavier) [inline srml/staking/src/lib.rs]: `active_ring = active_normal_ring + active_deposit_ring` useless checking

File: srml/staking/src/lib.rs (hunk 2 above)
Comment (yanganto) [reply srml/staking/src/lib.rs]: Thanks.

File: srml/staking/src/lib.rs (hunk 1 above)
Comment (yanganto) [reply srml/staking/src/lib.rs]: Thanks.

File: srml/staking/src/lib.rs (hunk 3 above)
Comment (yanganto) [reply srml/staking/src/lib.rs]: Thanks.

**Comment by yanganto** [comment]
Thanks for the review.

**Pull request merged**